qiskit/primitives/base/base_sampler.py

```diff
@@ -200,3 +257,33 @@ def parameters(self) -> tuple[ParameterView, ...]:
             List of the parameters in each quantum circuit.
         """
         return tuple(self._parameters)
+
+
+BaseSampler = BaseSamplerV1
+
+
+class BaseSamplerV2:
+    """Sampler base class version 2.
+
+    A Sampler returns samples of quantum circuit outputs.
+
+    A Sampler implementation must treat the :meth:`.run` method ``shots=None`` kwarg
+    as using a default ``shots`` value.  The default value and methods to set it can
+    be determined by the Sampler implementor.
+    """
+
+    @abstractmethod
+    def run(self, pubs: Iterable[SamplerPubLike], shots: int | None = None) -> Job:
```

Comment:
I suggest returning `PrimitiveJob` designated for Primitives. See #11552 for ongoing discussion.

Reply:
Since this will require adding the BasePrimitiveJob class, which isn't included in PR atm, can we make this change in a followup PR?

Reply:
It just merged